### PR TITLE
Fixes LoadBalancerFeignClient double loadbalancing

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**

### CODE REVIEW

**Comment by OlgaMaciaszek** [comment]
Fixes gh-1610.

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/TraceLoadBalancerFeignClient.java [hunk 1]
```diff
@@ -95,6 +101,11 @@ public Response execute(Request request, Request.Options options) throws IOExcep
 		}
 	}
 
+	private boolean delegateIsALoadBalancer() {
+		return getDelegate() instanceof LoadBalancerFeignClient
```
Comment (spencergibb) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/TraceLoadBalancerFeignClient.java:105]: Ug. I guess there's nothing better. :man_shrugging:

Comment (OlgaMaciaszek) [reply spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/TraceLoadBalancerFeignClient.java:105]: Yes, not sure we can do it any better way. In `master` we will only have to check against one class, at least.

**Comment by adriancole** [comment]
LGTM

**Pull request merged**